packages/@react-spectrum/color/src/ColorSlider.tsx

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {classNames} from '@react-spectrum/utils';
+import {Color, ColorSliderState, useColorSliderState} from '@react-stately/color';
+import {ColorChannel, ColorSliderProps} from '@react-types/color';
+import {ColorThumb} from './ColorThumb';
+import {Direction} from '@react-types/shared';
+import {Flex} from '@react-spectrum/layout';
+import {Label} from '@react-spectrum/label';
+import React, {useRef, useState} from 'react';
+import styles from '@adobe/spectrum-css-temp/components/colorslider/vars.css';
+import {useColorSlider} from '@react-aria/color';
+import {useFocus, useFocusVisible} from '@react-aria/interactions';
+import {useLocale} from '@react-aria/i18n';
+import {useProviderProps} from '@react-spectrum/provider';
+
+function getDisplayColor(value: Color, channel: ColorChannel): Color {
+  switch (channel) {
+    case 'hue':
+      return new Color(`hsl(${value.getChannelValue('hue')}, 100%, 50%)`);
+    case 'lightness':
+      value = value.withChannelValue('saturation', 0);
+    case 'brightness':
+    case 'saturation':
+    case 'red':
+    case 'green':
+    case 'blue':
+      return value.withChannelValue('alpha', 1);
+    case 'alpha': {
+      return value;
+    }
+    default:
+      throw new Error('Unknown color channel: ' + channel);
+  }
+}
+
+function getBackground(state: ColorSliderState, channel: ColorChannel, vertical: boolean, direction: Direction) {
```

Comment:
I think this function makes sense in aria. Doesn't seem to spectrum-specific to me...